internal/platform/trace/trace.go

```diff
@@ -0,0 +1,46 @@
+package trace
+
+import (
+	"context"
+	"log"
+
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/exporters/trace/zipkin"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+// InitTracer creates a new trace provider instance and registers it as global trace provider.
+func InitTracer(ServiceName string, LocalEndpoint string, ReporterURI string, logger *log.Logger) error {
+	// Create Zipkin Exporter
+	exporter, err := zipkin.NewExporter(
+		ReporterURI,
+		ServiceName,
+		zipkin.WithLogger(logger),
+	)
+	if err != nil {
+		return err
+	}
+
+	// For demoing purposes, always sample. In a production application, you should
+	// configure this to a trace.ProbabilitySampler set at the desired
+	// probability.
+	tp, err := sdktrace.NewProvider(
+		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
+		sdktrace.WithBatcher(exporter,
+			sdktrace.WithBatchTimeout(5),
+			sdktrace.WithMaxExportBatchSize(10),
+		),
+	)
+	if err != nil {
+		return err
+	}
+	global.SetTraceProvider(tp)
+	return nil
+}
+
+// NewSpan start a span with trace
+func NewSpan(ctx context.Context, name string) context.Context {
+	ctx, span := global.Tracer("main").Start(ctx, name)
+	defer span.End()
```

Comment:
This ends the span immediately? Doesn't think call need to be made by the caller on the returned context?

Reply:
I cloned your fork and ran the code. I am getting full spans. I don't understand how this could be working?

Reply:
Cloning your repo is not giving me this code. I forgot to switch branches.

Reply:
Ran the correct code. That call to End needs to happen by the caller. Please make that change.

Reply:
I'm making the change on your fork and I will try to submit a PR.

Reply:
I means don't be scattered anywhere for go.opentelemetry.io/otel/sdk/trace import, so using      NewSpan func, I modify that two return value.

Reply:
I am about to push some code to new.  Removed the New function, it was not necessary.

Reply:
I have a picture of the zipkin report I can show you before / after.

Reply:
ok, you do

Reply:
I didn't fork your fork. Can I have rights on this fork?

Reply:
I think you were right because your trace shows individual times per span. How does that work?

Reply:
I am very very confused. Sorry.

func NewSpan(ctx context.Context, name string) context.Context {
	ctx, span := global.Tracer("main").Start(ctx, name)
	defer span.End()
	return ctx
}
My guess is your trace is showing how long it takes to execute those two lines of code?

Reply:
That can't be right. I have code I think is right. I can push it to your repo or maybe what I will do it merge this into service. Then you can look at it?

Reply:
sales-api 
......687μs : internal.platform.web
 sales-api 
......2μs : internal.mid.logger
 sales-api 
......1μs : internal.mid.errors
 sales-api 
......1μs : internal.mid.metrics
 sales-api 
......13μs : handlers.check.health
sales-api 
......
I run curl -H "Authorization: Bearer ${TOKEN}" http://localhost:3000/v1/health show above, defer statement execute at function stack finished.

Reply:
> I am very very confused. Sorry.
> 
> func NewSpan(ctx context.Context, name string) context.Context {
> ctx, span := global.Tracer("main").Start(ctx, name)
> defer span.End()
> return ctx
> }
> My guess is your trace is showing how long it takes to execute those two lines of code?

this is wrong...span scope becomes smaller

Reply:
Ok, cool. I am going to move the changes to my repo, then you can take a look. I appreciate the work on this.

Reply:
Take a look at this

https://github.com/ardanlabs/service/pull/123

Reply:
you right for this pr, Probability property in Zipkin struct  has discarded

Reply:
Check the other PR moving forward.